src/graphql/data/SearchTokens.ts

```diff
@@ -0,0 +1,97 @@
+import { WRAPPED_NATIVE_CURRENCY } from 'constants/tokens'
+import gql from 'graphql-tag'
+import { useMemo } from 'react'
+
+import { Chain, SearchTokensQuery, useSearchTokensQuery } from './__generated__/types-and-hooks'
+import { chainIdToBackendName } from './util'
+
+gql`
+  query SearchTokens($searchQuery: String!) {
+    searchTokens(searchQuery: $searchQuery) {
+      decimals
+      name
+      chain
+      standard
+      address
+      symbol
+      market(currency: USD) {
+        price {
+          value
+          currency
+        }
+        pricePercentChange(duration: DAY) {
+          value
+        }
+        volume24H: volume(duration: DAY) {
+          value
+          currency
+        }
+      }
+      project {
+        id
+        logoUrl
+        safetyLevel
+      }
+    }
+  }
+`
+
+export type SearchToken = NonNullable<NonNullable<SearchTokensQuery['searchTokens']>[number]>
+
+function isMoreRevelantToken(current: SearchToken, existing: SearchToken | undefined, searchChain: Chain) {
+  if (!existing) return true
+
+  // Always priotize natives, and if both tokens are native, prefer native on current chain (i.e. Matic on Polygon over Matic on Mainnet )
+  if (current.standard === 'NATIVE' && (existing.standard !== 'NATIVE' || current.chain === searchChain)) return true
+
+  // Prefer tokens on the searched chain, otherwise prefer mainnet tokens
+  return current.chain === searchChain || (existing.chain !== searchChain && current.chain === Chain.Ethereum)
```

Comment:
this logic seems to be a little faulty. Searching for `eth` on arbitrum, I only get native wrapped in the current chain. Compared to current behaviour where most tokens are on the current chain
new:
![Screen Shot 2023-01-10 at 09 11 47 ](https://user-images.githubusercontent.com/11512321/211621062-b078b2bb-d7d9-4156-9543-4749ae2bef7c.png)
old:
![Screen Shot 2023-01-10 at 09 11 53 ](https://user-images.githubusercontent.com/11512321/211625830-8fca5b2d-45cf-4689-9216-fbb7c018263a.png)

Reply:
Thanks for catching, looks like its a data issue. Searching "wrapped" does return arbitrum weth... pinging the data team

Reply:
It seems like this is somewhat of a one-off. Arb WETH's on-chain name is "Wrapped Ethereum", while the others are "Wrapped Ether", so it has a weaker match. I don't think this is hugely important, because if a user is looking for WETH they're probably more likely to type "weth" or "wrapped"

Reply:
For Arbitrum searching rEth shows the chain specific version below the mainnet
<img width="543" alt="Screen Shot 2023-02-05 at 20 33 39 " src="https://user-images.githubusercontent.com/11512321/216884431-0dd6c28a-89ed-4bfa-bc21-76c4fab836c5.png">

Reply:
Another example is Uni on Optimism
<img width="996" alt="Screen Shot 2023-02-05 at 20 40 58 " src="https://user-images.githubusercontent.com/11512321/216885212-bc093e81-b3a2-42f6-ba8c-ff5fb82ed78b.png">